dkmanager_trainner/dkmanager_trainer/services/ai_trainer_worker_service.py

```diff
@@ -34,17 +48,45 @@ def __init__(self, api_origin: Optional[str] = None,
 
         self.start_services()
 
+    @staticmethod
+    def get_ai_worker_config() -> AIWorkerConfig:
+        """
+        :return: Current AI Worker configuration.
+        """
+        if Path(WORKER_CONFIG_LOCATION).exists():
+            return AIWorkerConfig.parse_file(WORKER_CONFIG_LOCATION)
+
+        return None
+
+    def create_ai_worker(self) -> Worker:
+        """
+        Create the AI Worker
+        :return: The newly created AI Worker
+        """
+        worker_create_request = WorkerCreate(type=WorkerType.AI_TRAINER, state=WorkerState.STOPPED)
+        worker = self._api.create_worker(worker_create_request)
+
+        # Save worker AI as configuration file
+        self.logger.info('Saving newly created worker id %i, for this AI Trainner in %s',
+                         worker.worker_id,
+                         WORKER_CONFIG_LOCATION)
+        Path(WORKER_CONFIG_LOCATION).write_text(AIWorkerConfig(worker_id=worker.worker_id).json())
+
+        return worker
+
     def create_or_get_worker(self) -> Worker:
         """
         Create IA worker if needed.
         """
-        ai_workers = self._api.get_workers(worker_type=WorkerType.AI_TRAINER)
+        worker = None
+
+        ai_worker_config = AiTrainerWorkerService.get_ai_worker_config()
+        if ai_worker_config is not None:
+            worker = self._api.get_worker(ai_worker_config.worker_id)
 
-        if len(ai_workers) > 0: # We assume there is only one IA Worker at all
-            worker = ai_workers[0]
+        if worker is not None:  # We assume there is only one IA Worker at all
```

Comment:
Isn't this comment wrong ?

Reply:
Comment it wrong yes,  updated here : 88c36e4